airflow/models/dag.py

```diff
@@ -361,8 +361,7 @@ def __repr__(self):
         return f"<DAG: {self.dag_id}>"
 
     def __eq__(self, other):
-        if type(self) == type(other) and self.dag_id == other.dag_id:
-
+        if type(self) == type(other):
```

Comment:
From what I can remember, it's optimization so we can check the most common case much faster.

Reply:
See: https://github.com/apache/airflow/commit/36e76fa08912a5f35eee940873814e6204f0873c

Reply:
That optimisation may not make sense given two reasons below:
- `all()` function has the "exit fast" property, i.e., whenever there is any `False` element, it will return `False` immediately, rather than traversing all elements in the iterable. [reference](https://docs.python.org/3/library/functions.html#all)
- For `dag` model, `dag_id` is the 1st element in `_comps`; For `BaseOperator`, `task_id` is the 1st element in `_comps`.

So after the change I make here, there should be zero impact on the performance (actually it improves the performance very minorly: it helps avoid comparing `dag_id` in `dag` model's `__eq__`for two times. Similar for `BaseOperator`).

Kindly let me know if it makes sense to you?

Reply:
SGTM.